chain/result.go

```diff
@@ -73,15 +60,6 @@ func UnmarshalResult(p *codec.Packer) (*Result, error) {
 	}
 	result.Consumed = consumed
 	result.Fee = p.UnpackUint64(false)
```

Comment:
I think we should also check that there are no bytes left to read here (seems I forgot to add that check previously)...it would've caught the bug I commented on above.